src/test/librbd/fsx.cc

```diff
@@ -2120,9 +2240,14 @@ test(void)
 	if (!quiet && testcalls < simulatedopcount && testcalls % 100000 == 0)
 		prt("%lu...\n", testcalls);
 
+	wsf = ws_factors[get_random() % (sizeof(ws_factors) / sizeof(ws_factors[0]))];
+	data_size = size / wsf;
+
 	offset = get_random();
-	if (randomoplen)
-		size = get_random() % (maxoplen + 1);
+	if (randomoplen) {
+		data_size = get_random() % (maxoplen / wsf);
```

Comment:
This change would affect all other ops and align them based on writesame limits. I would suggest following the example of ```writebdy``` and tweak the size directly within the ```dowritesame``` method.